src/scripts/Controller.js

```diff
@@ -1200,19 +1234,52 @@ export default class SitemapController {
 		this._editSelector(selector, sitemap);
 	}
 
-	initConfirmActionPanel(action) {
+	initConfirmActionPanel(action, extraExtion = () => {}) {
 		$('#confirm-action-modal').remove(); // remove old panel
 		$('#viewport').after(ich.ActionConfirm(action));
+		extraExtion();
+		const $confirmActionModal = $('#confirm-action-modal');
+		$confirmActionModal.modal('show');
 		Translator.translatePage();
 	}
 
-	showConfirmActionPanel(onConfirm) {
+	async askConfirmation() {
 		const $confirmActionModal = $('#confirm-action-modal');
-		$('#modal-submit').click(() => {
+		const confirm = await this.onConfirm();
+		if (confirm) {
 			$confirmActionModal.modal('hide');
-			onConfirm();
+			this.onConfirm = () => {};
+		} else {
+			$confirmActionModal.modal('show');
+		}
+	}
+
+	async copySitemap(button) {
+		let sitemap = $(button).closest('tr').data('sitemap');
+		this.initConfirmActionPanel({ action: 'copy_sitemap' }, async () => {
+			$('.modal-header').append(
+				'<form class="form-group"><input type="text" class="form-control" id="sitemapId" name="sitemapId" placeholder="sitemap_create_id"/></form>'
+			);
+			this.initCopySitemapValidation();
 		});
-		$confirmActionModal.modal('show');
+		$('#modal-message').show();
+		$('#modal-sitemap-id').text(sitemap._id);
+		this.onConfirm = async () => {
```

Comment:
I d suggests to use then chain here. That starts from modal creation -> ask confirmation -> delete/copy selector/sitemap.